gax-java/gax/src/main/java/com/google/api/gax/batching/BatcherImpl.java

```diff
@@ -241,12 +250,15 @@ public ApiFuture<ElementResultT> add(ElementT element) {
 
     SettableApiFuture<ElementResultT> result = SettableApiFuture.create();
     synchronized (elementLock) {
-      currentOpenBatch.add(element, result, throttledTimeMs);
-    }
+      if (!currentOpenBatch.isEmpty()
+          && batchingDescriptor.shouldFlush(
+              currentOpenBatch.resource.add(newResource), elementThreshold, bytesThreshold)) {
+        sendOutstanding();
+      }
```

Comment:
I dont think the isEmpty check is necessary here. You want to flush sendOutstanding either way...if its empty, then it will be a no-op